lib/saml11.js

```diff
@@ -169,11 +170,17 @@ function addSubjectConfirmation(options, doc, randomBytes, callback) {
 
 function sign(options, sig, doc, callback) {
   var token = utils.removeWhitespace(doc.toString());
-  sig.computeSignature(token, options.xpathToNodeBeforeSignature);
+  if (options.xpathToNodeBeforeSignature) {
+    options.location = {
```

Comment:
`option.location` is not very descriptive in the context of a saml option. Maybe we should put all fields (prefix, attrs and location) inside a unique option (like `options.signatureSettings = {}`)